File: back-end/Domain/BacklogItemServices/ListQuery/DTOs/BacklogItemListGetRequest.cs
using System.Collections.Generic;

using Raven.Yabt.Database.Common.BacklogItem;
using Raven.Yabt.Domain.Common;

namespace Raven.Yabt.Domain.BacklogItemServices.ListQuery.DTOs;

public class BacklogItemListGetRequest : ListRequest<BacklogItemsOrderColumns>
{
	/// <summary>
	///		Request items with specified types (user story, task, etc.)		
	/// </summary>
	/// <remarks>
	///		The (enum | null)[] type helps to prevent  "The value '' is invalid." validation error.
	///		See the problem description at https://stackoverflow.com/q/55868883/968003
	/// </remarks>
	public BacklogItemType?[]? Types { get; set; }

	/// <summary>
	///		Request items with specified states (new, closed, etc.)		
	/// </summary>
	/// <remarks>
	///		The (enum | null)[] type helps to prevent  "The value '' is invalid." validation error.
	///		See the problem description at https://stackoverflow.com/q/55868883/968003
	/// </remarks>
	public BacklogItemState?[]? States { get; set; }
		
	public string[]? Tags { get; set; }

	public string? Search { get; set; }

	public string? AssignedUserId { get; set; }

	public CurrentUserRelations? CurrentUserRelation { get; set; }

	public BacklogItemModification? UserModification { get; set; }

	public IDictionary<string,string>? CustomField { get; set; }
}

File: front-end/src/app/core/models/backlog-item/list/BacklogItemListGetRequest.ts
import { BacklogItemState } from '@core/models/common/BacklogItemState';
import { BacklogItemType } from '@core/models/common/BacklogItemType';
import { ListRequest } from '@core/models/common/ListRequest';
import { CurrentUserRelations } from './CurrentUserRelations';

export class BacklogItemListGetRequest extends ListRequest {
	type: keyof typeof BacklogItemType | undefined;
	states: Array<keyof typeof BacklogItemState> | undefined;
	tags: string[] | undefined;
	search: string | undefined;
	assignedUserId: string | undefined;
	currentUserRelation: keyof typeof CurrentUserRelations | undefined;
}
